test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java

```diff
@@ -38,7 +38,9 @@ public class LongGCDisruption extends SingleNodeDisruption {
         // security manager is shared across all nodes and it uses synchronized maps internally
         Pattern.compile("java\\.lang\\.SecurityManager"),
         // SecureRandom instance from SecureRandomHolder class is shared by all nodes
-        Pattern.compile("java\\.security\\.SecureRandom") };
+        Pattern.compile("java\\.security\\.SecureRandom"),
+        // Lucene's WindowsFS is shared across nodes and contains some coarse synchronization
+        Pattern.compile("org\\.apache\\.lucene\\.tests\\.mockfile\\.WindowsFS") };
```

Comment:
I totally like the exception message of the build scan that tells you to `Please add this occurrence to the unsafeClasses list in [org.elasticsearch.test.disruption.LongGCDisruption].` !

From a brief overview I understand that the test should not suspend threads that may share locks with the caller thread, in order to simulate a GC. And it seems that some lucene sync objects for WindowsFS were involved.

Reply:
> the test should not suspend threads that may share locks with the caller thread

Almost - it's more that we should not suspend threads while they hold a lock which is shared between nodes. In this case all nodes share the filesystem instance, so by blocking one node while it holds the `WindowsFS#openFiles` mutex we effectively prevent the other nodes from doing I/O too.

I also considered giving each node its own `WindowsFS` instance instead, but that doesn't seem to be something that the (Lucene-derived) test framework supports today so it'd be quite complex to achieve and I don't think it's worth the effort.